src/js/covidMap.js

```diff
@@ -80,6 +80,89 @@ fluid.covidMap.extractPostcodes = function (rows, field) {
     return postcodeHash;
 };
 
+fluid.covidMap.relativePath = function (basePath, relativePath) {
+    var lastSlash = basePath.lastIndexOf("/");
+    return basePath.substring(0, lastSlash + 1) + relativePath;
+};
+
+fluid.covidMap.fileToRelative = function (fileName, suffix) {
+    var lastdotpos = fileName.lastIndexOf(".");
+    return fileName.substring(0, lastdotpos) + suffix;
+};
+
+/** Shorten some text if its length is excessive by replacing the portion between a prefix and suffix point with "..."
+ * @param {String} text - The text to be shortened
+ * @param {Integer} prefix - The prefix length of the text to be retained
+ * @param {Integer} suffix - The suffix length of the text to be retained
+ * @return {String} A shortened version of the supplied text
+ */
+fluid.covidMap.shortenText = function (text, prefix, suffix) {
+    return text.length > prefix + suffix + 3 ?
+        text.substr(0, prefix) + "..." + text.substr(-suffix) : text;
+};
+
+/** Fetch a file relative to the one provided in "latest.json". Don't we wish we had i) general asynchrony, ii)
+ * Edwardian "multiple returns".
+ * @param {String} latestUrl - A url to a "latest.json" file in the format produced e.g. in covid-assessment-centres.
+ * @param {String} [suffix] - optional - if provided, replace the suffix of the file found within "latest" with this one
+ * @return {Promise<Object>} A promise for the JSON contents of the fetched file
+ */
+fluid.covidMap.fetchLatestRelative = function (latestUrl, suffix) {
+    var togo = fluid.promise();
+    fetch(latestUrl).then(function (response) {
+        response.json().then(function (data) {
+            console.log("Got data ", data);
+            var relative = fluid.covidMap.relativePath(latestUrl, data.fileName);
+            var toFetch = suffix ? fluid.covidMap.fileToRelative(relative, suffix) : relative;
+            fetch(toFetch).then(function (response2) {
```

Comment:
This logic will first fetch `latestUrl` then fetch `latestUrl+suffix` so the provenanceMap fetch at line 135-139 will cover the fetch of `latestUrl` at line 131-134. Would it be more efficient to fetch only one of them in conditions of: 1. fetch `latestUrl` when the suffix is not provided; 2. fetch `latestUrl+suffix` when the suffix is provided

Reply:
It would be more efficient but there's no clear way to express it in the rather inflexible Infusion 3.x ResourceLoader idiom. I've written up the problem here - https://wiki.fluidproject.org/display/fluid/Bestiary+of+Reuse+Failures#BestiaryofReuseFailures-Infusion'sResourceLoadermodel